js/ui/grid_core/ui.grid_core.column_chooser.js

```diff
@@ -43,12 +41,11 @@ const processItems = function(that, chooserColumns) {
                 allowHiding: column.allowHiding,
                 expanded: true,
                 id: column.index,
-                disabled: false,
-                disableCheckBox: column.allowHiding === false,
+                disabled: column.allowHiding === false,
                 parentId: isDefined(column.ownerBand) ? column.ownerBand : null
             };
 
-            if(isSelectMode) {
+            if(isSelectMode && !(isRecursive && column.hasColumns)) {
```

Comment:
Minor comment.

Let's move condition in brackets into a separate variable:

```suggestion
            const isRecursiveWithColumns = isRecursive && column.hasColumns;
            if(isSelectMode && !isRecursiveWIthColumns) {
```